### ISSUE TITLE
Add nearest property for tooltip.

### ISSUE THREAD

**Issue by davidwales** [vega/vega-lite#6851]
## Suggestion

Add `nearest` property to `tooltip`:

```json
{
  "mark": {"type": "line", "tooltip": {"content": "encoding", "nearest": true}},
  "encoding": {"x": {"field": "a"}, "y": {"field": "b"}}
}
```

## Motivation

I have enabled tooltips on a [line chart](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JACyUAVhDYA7EABoQAdxoATemhBQ5mJDVlwATooxwaZLFp2y9B46ZsAzC2lBkjbBjjcgXgwS0AYjgAVjCwkABfJSQjKF9-QPQggCYmVKgAZgBOaNijOBRUUCTg9Mzc-JBhQ18ITE8AazhyjOy8mJAcJE0SkAbm1pSKjurCqExEmgC2ys6lCBgkHGHQQbYWubGuiABPBCY2ZNKZ5JA09qquoyRZMjXtXvI2Iz20AG0L1NSADiykKlTEEAAwgrJwABsKCUQSQAEYoBFgQAxAAiACEAKLwrKo1IAdkJWOBWIAglkUZTgalQmScliQaicgSySDfsCxPCMSiMRzYaF4b80QBhMnAnJiSEgsn8i6QlEEtG-MTAtHw0Vk0IgAC6US6mDogmGIBFDAQDEEvRopAABCKAPIANQAkmj+PCcvakBA4BBbUw9raADIAcTJtoMtoAyl56LaAOp+zDGWQxvZqQzvRZwY2TdjyfpqZ6+TB7VZaAwpozEJCBJRMAxqLTQOt+0xwWQ6NQGMgQT4gAAeuq6MDYpBM-TLFfQEF7xtMci0CC8vvHDiUhjiybQjQYcH1SkKMkEpHqUDbvne6CbcFWsjUnamh5Axb0vgYRnOWFw+CIxeIdRlDbCBqDoRgWHYAhZAgZR+B0QC1A9HJ+DfJACAQ9RkPgn0-X4QN+FkNhDRcC9DTkVDniENgyILfg7iQ4QWkEPZ+BkT8kX4NgnH4AwnDgfM5EoKAIGIapGjuCAnFeBBPlOQRq18Bi5L8Ghc2bdBBBgNQAH1UhBeEQR02QkAQVolDgABHBg6y0WMGHjJMGlTdNMzgd4YlAIipn6Fx1K0QQKGM0y4E9DtrNs9AAAVOyMcDoiiPVPJAJAyA8PgUxUtgfHQHQGFsUwfS0N84B0wKkB0vKCsSpQPC8HBAwHbyaFI605B0krTHK4KzLCvUFFAVQHzYZQspygZzVMPzBA019njKoKqqmWJ+1y80Ovm7qluiHVas8bxGtQL5upM3q8l2vxbjMz5ZEtQQFBBPULuQIwmlLcsTVqcyQEwNhjkNHK9wPCyuzYHt7l8YcpxoMyAFVZDoLQ9iKIwV1sbSSyUabZua1ryNkDbMqUQ1MAXdA0Uxn6Pq0FMEBwV5bK6a9Tn8taEEJ0qtq8AriaNE0zQtK1DTtEVcP7YnqfQCLbDoa0zy6HRBFeaZWZqILTtCvIJZnEAiIQAxbN50mTTDcUlGNB4H18V41IK9Bfp8JRFfNGC0DEfV9SAA).
However, the tooltip only appears when you hover over an actual data point, rather than over the line itself.
The data points in a line are very small and often far apart, so having something like `nearest` might make it easier to activate the tooltip.

**Comment by JeffreyPalmer** [comment]
Is there a way to do this manually in the meantime?

**Comment by eflister** [comment]
- you would only want the tooltip when actually touching the line, as opposed to a tooltip for whatever line you are nearest
- a slight enhancement would be to consider the datapoints on both sides of the contact point, if they disagree at a tooltip field, show something like "field: value 1 - value 2" or maybe an interpolation btw the values

**Comment by domoritz** [comment]
The nearest computation could have a limit above which we don't show the tooltip.

**Comment by joelostblom** [comment]
I agree that a nearest property for the tooltip would be helpful. Currently it can be hard to hit the exact pixel of e.g. a line chart corner to show the tooltip. Especially together with a distance limit (https://github.com/vega/vega-lite/issues/4262) and some sort of indication of which data point the tool tip belongs to (e.g. with a call out wedge/arrow) would make the tooltips easier to use. I think how this is handled in plotly is quite intuitive in terms of both the distance of activation and the little arrow pointing at the data point.

https://github.com/vega/vega-lite/assets/4560057/13ac4f90-1bee-4857-b53a-66a2ebd44b8d

**Comment by rgowtham** [comment]
It would help to have this feature implemented - currently there seems to be workarounds that can achieve this as per this discussion - https://github.com/vega/altair/discussions/3597. But it would be helpful to have this implemented directly
